tests/datasource/test_new_datasource_with_runtime_data_connector.py

```diff
@@ -368,3 +383,343 @@ def test_get_batch_definitions_and_get_batch_basics(
         )
     )
     assert batch.batch_request == {}
+
+
+####################################
+# Tests with data passed in as query
+####################################
+
+
+@pytest.fixture
+def datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine(sa):
+    db_file = file_relative_path(
+        __file__,
+        os.path.join("..", "test_sets", "test_cases_for_sql_data_connector.db"),
+    )
+
+    basic_datasource: Datasource = instantiate_class_from_config(
+        yaml.load(
+            f"""
+    class_name: Datasource
+
+    execution_engine:
+        class_name: SqlAlchemyExecutionEngine
+        connection_string: sqlite:///{db_file}
+
+    data_connectors:
+        test_runtime_data_connector:
+            module_name: great_expectations.datasource.data_connector
+            class_name: RuntimeDataConnector
+            batch_identifiers:
+                - pipeline_stage_name
+                - airflow_run_id
+                - custom_key_0
+        """,
+        ),
+        runtime_environment={"name": "my_datasource"},
+        config_defaults={"module_name": "great_expectations.datasource"},
+    )
+    return basic_datasource
+
+
+def test_datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine_self_check(
+    datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine, sa
+):
+    db_file = file_relative_path(
+        __file__,
+        os.path.join("..", "test_sets", "test_cases_for_sql_data_connector.db"),
```

Comment:
Thoughts on moving this `db_file` to a separate fixture? A matter of taste I think what you have is definitely fine.
```suggestion
@pytest.fixture
def db_file():
    return file_relative_path(__file__, os.path.join("..", "test_sets", "test_cases_for_sql_data_connector.db"))
    
@pytest.fixture
def datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine(db_file, sa):
    basic_datasource: Datasource = instantiate_class_from_config(
        yaml.load(
            f"""
    class_name: Datasource

    execution_engine:
        class_name: SqlAlchemyExecutionEngine
        connection_string: sqlite:///{db_file}

    data_connectors:
        test_runtime_data_connector:
            module_name: great_expectations.datasource.data_connector
            class_name: RuntimeDataConnector
            batch_identifiers:
                - pipeline_stage_name
                - airflow_run_id
                - custom_key_0
        """,
        ),
        runtime_environment={"name": "my_datasource"},
        config_defaults={"module_name": "great_expectations.datasource"},
    )
    return basic_datasource


def test_datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine_self_check(
    db_file, datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine, sa
):
    # Rest of the test
```